core/harambe_core/parser/type_price.py

```diff
@@ -0,0 +1,79 @@
+import re
+from typing import Any, Optional, Union
+
+from price_parser import Price
+from pydantic import BeforeValidator
+from typing_extensions import Annotated, Literal
+
+from harambe_core.parser.constants import CURRENCY_MAP, PRICE_NOT_AVAILABLE_PHRASES
+from harambe_core.parser.type_currency import ParserTypeCurrency
+
+# this will capture only the amount
+number_pattern = re.compile(r"-?.?\d[\d,-\.]*")
+
+PriceOutputType = dict[
```

Comment:
make this a type dict:

class PriceOutputType(TypedDict)
  currency: str
  raw_currency: str 
  ammount: flot